issue_processing/jira_issue_processing.py

```diff
@@ -70,52 +70,82 @@ def run():
     # get person folder
     # __psrcdir = os.path.abspath(os.path.join(args.resdir, __conf["repo"] + "_proximity", "conway"))
 
-    # 1) load the list of issues
-    issues = load_xml(__srcdir)
-    # 1b) load the list of persons
+    # load the list of persons
     persons = load_csv(__srcdir_csv)
-    # 2) re-format the issues
-    issues = parse_xml(issues, persons, args.skip_history)
-    # 3) load issue information via api
-    if not args.skip_history:
-        load_issue_via_api(issues, persons, __conf["issueTrackerURL"])
-    # 4) update user data with Codeface database
-    # mabye not nessecary
-    issues = insert_user_data(issues, __conf)
-    # 5) dump result to disk
-    print_to_disk(issues, __resdir)
-    # 6) export for Gephi
-    print_to_disk_gephi(issues, __resdir)
-    # 7) export for jira issue extraction to use them in dev-network-growth
-    print_to_disk_extr(issues, __resdir)
-    # 8) dump bug issues to disk
-    print_to_disk_bugs(issues, __resdir)
+
+    # load the xml-file list
+    file_list = [f for f in os.listdir(__srcdir) if os.path.isfile(os.path.join(__srcdir, f))]
```

Comment:
There are usually only XML files in the folder. But we had already the situation where XML files were broken and couldn't be read. @clhunsen, can we check the correctness of a XML file?

Reply:
Honestly, I do not have clue right now. I guess, when you call the parser on a malformed or non-existent XML file, it will throw an exception. At least, this is worth a try.

If my assumption is true, the handling is easy: We handle the exceptions in the function `load_xml`, return `None` in the error case, and handle this case then in the outer for-loop by skipping the current file (while giving a log message!).

Reply:
By skipping a malformed file, I did not mean to ignore it. Actually, with handling an exception, you gain two advantages: First, you get incremental (but well-formed) data to work with in further analysis steps. Second, you can print a complete list of malformed files at the end – without running the extraction process several times, each time waiting for the next error.